src/main/java/at/hannibal2/skyhanni/features/inventory/DojoRankDisplay.kt

```diff
@@ -0,0 +1,74 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class DojoRankDisplay {
+
+    private val config get() = SkyHanniMod.feature.inventory
+    private var display = emptyList<String>()
+    private val patternGroup = RepoPattern.group("inventory.dojo.rankdisplay")
+    private val testNamePattern by patternGroup.pattern("name", "(?<color>§\\w)Test of (?<name>.*)")
+    private val testRankPattern by patternGroup.pattern("rank", "(?:§\\w)+Your Rank: (?<rank>§\\w.) §8\\((?<score>\\d+)\\)")
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        config.dojoRankDisplayPosition.renderStrings(display, posLabel = "Dojo Rank Display")
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        if (event.inventoryName != "Challenges") return
+        val newDisplay = mutableListOf<String>()
+        var totalScore = 0
+        for (stack in event.inventoryItems.values) {
+            for (line in stack.getLore()) {
+                val name = stack.displayName ?: return
+                testRankPattern.matchMatcher(line) rank@{
+                    testNamePattern.matchMatcher(name) name@{
+                        val testColor = this@name.group("color")
+                        val testName = this@name.group("name")
+                        val rank = this@rank.group("rank")
+                        val score = when (val s = this@rank.group("score").toInt()) {
```

Comment:
do not duplicate the group function with parameter "score", use a variable for this. Also dont return the score in this when check but the color code instead. Also maybe dont use a when but rather a if instead.